bin/run-tests

```diff
@@ -26,6 +26,8 @@ for D in exercises/*; do
           make clean >> /dev/null;
           if make | grep FAIL: ; then
             exit 1
+          elif [ ${PIPESTATUS[0]} -ne 0 ]; then
```

Comment:
I tried that a while ago and I could never get it to actually work.  I think I just figured out why (from the Bash Reference Manual):
> The shell does not exit if the command that fails is part of the command list immediately following a while or until keyword, part of the test in an if statement ...

In this case, it's part of an `if` statement, so it doesn't exit.  Although it couldn't hurt to add `set -e` to catch any other unforeseen errors.